src/cursor/cur_hs.c

```diff
@@ -938,10 +939,9 @@ __curhs_insert(WT_CURSOR *cursor)
 
 #ifdef HAVE_DIAGNOSTIC
     /* Do a search again and call next to check the key order. */
-    WT_WITH_PAGE_INDEX(session, ret = __curhs_search(cbt, false));
+    WT_WITH_TXN_ISOLATION(session, WT_ISO_READ_UNCOMMITTED, ret = file_cursor->search(file_cursor));
```

Comment:
Why do some calls to __curhs_search use a WT_WITH_PAGE_INDEX macro and others don't? It seems like something that should be completely consistent.

Reply:
@agorrod  I checked all the usage of __curhs_search. They are all wrapped with WT_WITH_PAGE_INDEX. Can you point me to which usage?

Reply:
I checked that all the usage of __curhs_search are wrapped with WT_WITH_PAGE_INDEX. I think you are confused with some function calls that prefix with "__curhs_search", such as "__curhs_search_near_helper"